server/src/controllers/load.rs

```diff
@@ -0,0 +1,53 @@
+use super::util::respond;
+use crate::parsers::odt::ODTParser;
+use tiny_http::Request;
+
+/// Handles a request for loading a file
+pub fn load_controller(mut request: Request) {
+    let req_reader = request.as_reader();
+    let mut body_bytes: Vec<u8> = Vec::new();
+    if let Err(e) = req_reader.read_to_end(&mut body_bytes) {
+        respond(request, e.to_string(), true);
+        return;
+    }
+
+    let body_str = std::str::from_utf8(&body_bytes);
+    if let Err(e) = body_str {
+        respond(request, e.to_string(), true);
+        return;
+    }
+
+    let filepath = body_str.unwrap();
+    let file = std::path::Path::new(&filepath);
+    if !file.exists() {
+        respond(request, format!("{:?}", std::fs::metadata(file)), true);
+        return;
+    }
+
+    let extension = filepath.split('.').last();
+    match extension {
+        //pick a parser depending on the file extension
```

Comment:
For consistency with [Rust style RFC](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/guide.md#comments):

> When using line comments there should be a single space after the opening sigil.

```suggestion
        // Pick a parser depending on the file extension
```